spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDefinitionController.java

```diff
@@ -141,6 +141,9 @@ public void save(@RequestParam("name") String name,
 					@RequestParam(value = "deploy", defaultValue = "false")
 					boolean deploy) {
 		StreamDefinition stream = new StreamDefinition(name, dsl);
+		if (this.repository.findOne(stream.getName()) != null) {
```

Comment:
When we shift to Spring Data JPA, the save method is going to do an persist (if new) or merge if it exists already.  So if we want to unsure that you can't save a stream/task def more than once, the logic is going to need to shift back into the controllers for stream/task - or introduce a service layer.  I'll make a note of this on the issue for when we switch to using JPA. https://github.com/spring-cloud/spring-cloud-dataflow/issues/311.